### DISCUSSION TITLE
Java/Gradle documentation is incorrect/misleading

### DISCUSSION THREAD

**Discussion by drabe8** [renovatebot/renovate#14674] [Suggest an Idea]
At https://docs.renovatebot.com/java/, we read that renovate can update:
```
dependencies whose version is defined in a *.properties file
```

It doesn't tell you that it won't look in all properties files by default! After adding this packageRule:
```
  "gradle": {
    "fileMatch": ["gradle_versions.properties"]
  }
```
I see that my properties file is matched; however, the dependencies that use properties from that file are not considered for update.

After digging into the code, I found the "isPropsFile" code at https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/gradle/utils.ts#L121
```
export function isPropsFile(path: string): boolean {
  const filename = upath.basename(path).toLowerCase();
  return filename === 'gradle.properties';
}
```

With that new insight, I moved my version properties into gradle.properties, and renovate was able to consider my dependencies for update.

I discovered this while trying to recover from the removal of deepExtract, which had been working very nicely for us. Without deepExtract, it seems like we have to restructure our gradle projects to fit Renovate's capabilities. As such, accurate and complete documentation is more important than ever! Thank you!

**Comment by rarkins** [comment]
PRs for documentation are welcome
